packages/request-node/src/storageUtils.ts

```diff
@@ -29,33 +28,33 @@ export function getEthereumStorage(
   };
 
   // Initializes web3 connection object
-  let provider: HDWalletProvider;
-  if (config.getStorageWeb3ProviderUrl().match('^wss?://.+')) {
-    provider = new HDWalletProvider({
-      mnemonic,
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-ignore
-      providerOrUrl: new Web3WsProvider(config.getStorageWeb3ProviderUrl(), {
-        clientConfig: {
-          keepalive: true,
-          keepaliveInterval: 10000, // ms
-        },
-        // Enable auto reconnection
-        reconnect: {
-          auto: true,
-          delay: 3000, // ms
-          maxAttempts: 5,
-          onTimeout: false,
-        },
-      }),
-    });
+
+  let provider: providers.Provider;
+  const url = config.getStorageWeb3ProviderUrl();
+  if (url.match('^wss?://.+')) {
+    provider = new providers.WebSocketProvider(url);
+    // FIXME previous WS config
```

Comment:
@alexandre-abrioux any idea what we should do here?

Reply:
You can do this:
```ts
provider = new providers.Web3Provider(
  new Web3WsProvider(url, {
    //  previous WS config
  }),
)
```
; or, if we want to completely remove web3js dependencies, I can take care of it in another PR, for now just remove the `if (wss)` block or throw an "unsupported" error) as `ethers.js` does not handle auto-reconnect, cf. https://github.com/ethers-io/ethers.js/issues/1053

Reply:
the point is to remove web3 completely yes

Reply:
We should use this solution then: https://github.com/ethers-io/ethers.js/issues/1053#issuecomment-808736570 ,  we just need to abstract the provider in it's own module.